tests/System.Buffers.Primitives.Tests/BufferSegment.cs

```diff
@@ -99,12 +100,17 @@ public override string ToString()
                 return "<EMPTY>";
             }
 
-            var builder = new StringBuilder();
-            SpanLiteralExtensions.AppendAsLiteral(Memory.Span, builder);
-            return builder.ToString();
+            if (typeof(T) == typeof(byte))
+            {
+                var builder = new StringBuilder();
+                SpanLiteralExtensions.AppendAsLiteral(MemoryMarshal.AsBytes(Memory.Span), builder);
+                return builder.ToString();
+            }
+
+            return Memory.Span.ToString();
```

Comment:
Just checking, this will output the right thing (i.e. the actual string contents) if T = char, correct?

Reply:
Should- this was a convenience for the test code.